rust/arrow/src/compute/kernels/window.rs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Defines windowing functions, like `shift`ing
+
+use crate::compute::concat;
+use num::{abs, clamp};
+use std::sync::Arc;
+
+use crate::{
+    array::{make_array, ArrayData, PrimitiveArray},
+    datatypes::ArrowPrimitiveType,
+    error::Result,
+};
+use crate::{
+    array::{Array, ArrayRef},
+    buffer::MutableBuffer,
+};
+
+/// Shifts array by defined number of items (to left or right)
```

Comment:
I wonder if you can add a doc example here, something like https://github.com/apache/arrow/blob/master/rust/arrow/src/compute/kernels/take.rs#L62-L76 just to show the behavior (lifting / simplifying one of unit tests is probably a good level)

Reply:
It might also be good to specify what happens if you shift by more than the size of the array (e.g. that there is no error, just the entire array is returned as null)